From 14d2006ca6c0e2b54784b93560f09e0e19c0a270 Mon Sep 17 00:00:00 2001
From: Diego Biurrun <diego@biurrun.de>
Date: Fri, 15 Aug 2014 22:13:10 +0200
Subject: [PATCH 09/37] pcm: Drop unused variable from DECODE_PLANAR macro

---
 libavcodec/pcm.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/libavcodec/pcm.c b/libavcodec/pcm.c
index 9fadcb7..2985156 100644
--- a/libavcodec/pcm.c
+++ b/libavcodec/pcm.c
@@ -264,7 +264,6 @@ static av_cold int pcm_decode_init(AVCodecContext *avctx)
 #else
 #define DECODE_PLANAR(size, endian, src, dst, n, shift, offset)         \
     {                                                                   \
-        int av_unused n2;                                               \
         n /= avctx->channels;                                           \
         for (c = 0; c < avctx->channels; c++) {                         \
             samples = frame->extended_data[c];                          \
-- 
1.9.1

