From 6af2930222ee5d8ce19f3b999a78d85a3c457391 Mon Sep 17 00:00:00 2001
From: Diego Biurrun <diego@biurrun.de>
Date: Fri, 15 Aug 2014 22:13:11 +0200
Subject: [PATCH 10/37] pcm: Drop av_unused attribute from variable that is
 always used

---
 libavcodec/pcm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/pcm.c b/libavcodec/pcm.c
index 2985156..e06712e 100644
--- a/libavcodec/pcm.c
+++ b/libavcodec/pcm.c
@@ -253,7 +253,7 @@ static av_cold int pcm_decode_init(AVCodecContext *avctx)
 #if HAVE_BIGENDIAN
 #define DECODE_PLANAR(size, endian, src, dst, n, shift, offset)         \
     {                                                                   \
-        int av_unused n2;                                               \
+        int n2;                                                         \
         n /= avctx->channels;                                           \
         for (c = 0; c < avctx->channels; c++) {                         \
             samples = frame->extended_data[c];                          \
-- 
1.9.1

