From 480633c6c2e1434c981cc887c6d54d502e24d6d7 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michaelni@gmx.at>
Date: Mon, 1 Sep 2014 00:48:50 +0100
Subject: [PATCH 33/37] avcodec: fix missing doxygen comment marker

---
 libavcodec/avcodec.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/avcodec.h b/libavcodec/avcodec.h
index 14440fe..253e45a 100644
--- a/libavcodec/avcodec.h
+++ b/libavcodec/avcodec.h
@@ -912,7 +912,7 @@ enum AVPacketSideDataType {
      */
     AV_PKT_DATA_DISPLAYMATRIX,
 
-    /*
+    /**
      * This side data should be associated with a video stream and contains
      * Stereoscopic 3D information in form of the AVStereo3D struct.
      */
-- 
1.9.1

