From 59cc0812207e3c2e2f2e559d60d0a8f9296a8847 Mon Sep 17 00:00:00 2001
From: Reinhard Tartler <siretart@tauware.de>
Date: Sun, 10 Aug 2014 10:07:55 -0400
Subject: [PATCH 5/7] Prepare for 11_alpha2 Release

---
 RELEASE | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/RELEASE b/RELEASE
index 086e7dc..83512dd 100644
--- a/RELEASE
+++ b/RELEASE
@@ -1 +1 @@
-11_alpha1
+11_alpha2
-- 
1.9.1

