From ff24824a721576195c3b4a711e3ee2af900de795 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michaelni@gmx.at>
Date: Sun, 28 Sep 2014 00:46:50 +0100
Subject: [PATCH 06/10] h264: reset ret to avoid propagating minor failures

Unbreak 772d150a6e82542c06b0c251e73dd299d98d1027.

CC: libav-stable@libav.org
Bug-Id: 750 / 905753
Signed-off-by: Vittorio Giovara <vittorio.giovara@gmail.com>
---
 libavcodec/h264.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libavcodec/h264.c b/libavcodec/h264.c
index 1fd5e50..562b102 100644
--- a/libavcodec/h264.c
+++ b/libavcodec/h264.c
@@ -1679,6 +1679,7 @@ again:
             goto end;
     }
 
+    ret = 0;
 end:
     /* clean up */
     if (h->cur_pic_ptr && !h->droppable) {
-- 
2.1.0

